### ISSUE TITLE
Usernames involving special symbols (email, etc.) can't log in / view userpage

### ISSUE THREAD

**Issue by calculuschild** [naturalcrit/homebrewery#807]
Seems to occur occasionally. Creating this issue to centralize the related threads:

#662, #555, #547, #401, #349

**Comment by ericscheid** [comment]
Closing #764, re-opening this as not resolved(?)

**Comment by calculuschild** [comment]
Good catch. I probably closed this by accident when I linked it to the other duplicate.

**Comment by ericscheid** [comment]
> Clicking on my name in the top right corner does nothing but bring me to the same welcome example page thing with all the instructions on it.

> Currently, I seem to be unable to access my user profile. When I try and access my profile by clicking on my email in the top right corner, I'm sent back to the main page instead of my user page, meaning I can't access any of the brews that I've currently made or any I make in future. I've tried this on chrome and firefox and I've got the same issue on each browser.

----

However this link works, revealing a published brew: https://homebrewery.naturalcrit.com/user/lakestral@gmail.com

![image](https://user-images.githubusercontent.com/2605467/99152726-a1e74d00-26f7-11eb-9af2-2b979c8ccc42.png)

So, maybe usernames as emails work .. but that says nothing about other edge cases so leaving open.

Note however that the username *is* case-sensitive: https://homebrewery.naturalcrit.com/user/lakestral@GMAIL.COM

![image](https://user-images.githubusercontent.com/2605467/99152758-de1aad80-26f7-11eb-9da7-46541cf5c2ee.png)

We need more examples of failures.

**Comment by calculuschild** [comment]
I think the link generated in the profile button is url encoded, turning the @ symbol into %40 or whatever. We just need to make the button not do that.

**Comment by ericscheid** [comment]
https://github.com/naturalcrit/homebrewery/blob/master/client/homebrew/navbar/account.navitem.jsx#L23-L25

	return <Nav.item href={`/user/${global.account.username}`} color='yellow' icon='fa-user'>
		{global.account.username}
	</Nav.item>;

**Comment by 5e-Cleric** [comment]
https://github.com/naturalcrit/naturalcrit/issues/63 is also interesting to solve this and prevent similar issues.

**Comment by 5e-Cleric** [comment]
Isn't this fixed?

**Comment by G-Ambatte** [comment]
Ran into this issue today for user `AKG#3356`. When they clicked the "BREWS" button in the Account menu, they got the following:

<img width="664" height="381" alt="Image" src="https://github.com/user-attachments/assets/973342b9-6524-437d-b4db-17244796d9f0" />

However they reported that the URI encoded URL worked for them.

<img width="576" height="157" alt="Image" src="https://github.com/user-attachments/assets/5849f2d6-88b2-468c-96cc-d92ca272bc2f" />

I suspect that the solution is to `encodeURIComponent()` any text that is added to the URL, including any instances of `/user/${account.username}`. In the limited testing completed today, it appears that there are no issues with reading the encoded data from the URL, only with that data not being encoded correctly when it is added.
